site/src/pages/WorkspacePage/WorkspacePage.tsx

```diff
@@ -85,7 +91,7 @@ export const WorkspacePage: React.FC = () => {
               bannerSend({
                 type: "UPDATE_DEADLINE",
                 workspaceId: workspace.id,
-                newDeadline: dayjs(workspace.latest_build.deadline).utc().add(4, "hours"),
+                newDeadline: dayjs.min(deadline.add(4, "hours"), maxDeadline(workspace, template)),
```

Comment:
You don't have to change this here, but I'm wondering if we want to try to push logic like this into XState. Instead of one `UPDATE_DEADLINE` event, we could have multiple events that do calculations within them. The idea would be to try to get more logic out of components. Maybe a topic for Frontend Variety.

Reply:
Yeah, I was wondering that but didn't want to over-complicate the machine. :-)